dds_web/api/user.py

```diff
@@ -271,8 +271,7 @@ def add_to_project(whom, project, role, send_email=True):
             return {
                 "status": ddserr.AccessDeniedError.code.value,
                 "message": (
-                    "User Role should be either 'Project Owner' or "
-                    "'Researcher' to be added to a project"
+                    f"This user is registered as a '{role}'. Role should be either 'Project Owner' or 'Researcher' to be added to a project."
```

Comment:
Do we want to leak the role of the user?

Reply:
```suggestion
                    f"This user is registered as an incompatible role. Role should be either 'Project Owner' or 'Researcher' to be added to a project."
```

Reply:
How many roles are there? Is it not fairly obvious already?

Reply:
Maybe a non-issue. I have just become a bit paranoid.

Reply:
No it's a good point. Printing the role doesn't add much, I'll change it.

```suggestion
                    f"This user is already registered with an incompatible role. Role should be either 'Project Owner' or 'Researcher' to be added to a project."
```

Reply:
```
root@6f52f567f7ab:/code# dds user add unituser2@mailtrap.io --role Researcher -p someunit00001
     ︵ 
 ︵ (  )   ︵ 
(  ) ) (  (  )   SciLifeLab Data Delivery System 
 ︶  (  ) ) (    http://dds_backend:5000/ 
      ︶ (  )    Version 0.0.11 
          ︶
Current user: unituser_1
ERROR    ⚠️ Could not add user: This user is already registered with an incompatible role. Role should be either 'Project Owner' or 'Researcher' to be added to a project. ⚠️    

                           
root@6f52f567f7ab:/code# dds user add unituser2@mailtrap.io --role "Unit Personnel" -p someunit00001
     ︵ 
 ︵ (  )   ︵ 
(  ) ) (  (  )   SciLifeLab Data Delivery System 
 ︶  (  ) ) (    http://dds_backend:5000/ 
      ︶ (  )    Version 0.0.11 
          ︶
Current user: unituser_1
ERROR    ⚠️ Could not add user: This user is already registered with an incompatible role. Role should be either 'Project Owner' or 'Researcher' to be added to a project. ⚠️   
```

So these two give the same response, but I think the `This user is already registered with an incompatible role` can cause a bit of confusion.

Reply:
For
```
dds user add unituser2@mailtrap.io --role "Unit Personnel" -p someunit00001
```
it should just say something in the lines of "You cannot add a user with the role Unit Personnel to a project."